src/code/sched.c

```diff
@@ -165,90 +227,145 @@ OSScTask* func_800C89D4(SchedContext* sc, OSScTask* task) {
         return NULL;
     }
 
-    if ((sc->pendingSwapBuf2 != NULL ? sc->pendingSwapBuf2->swapBuffer : NULL) == task->framebuffer->fb1) {
+    // If the task's framebuffer is one of the pending swaps or NULL.
```

Comment:
I think half the ternary is dead code because of the `if (sc->pendingSwapBuf1 != NULL` above

Overall this function doesn't make sense to me

Reply:
It is, I mention this in a comment already

Reply:
I must be blind :see_no_evil: